### DISCUSSION TITLE
Block actual gamepad buttons?

### DISCUSSION THREAD

**Discussion by ghost** [AntiMicroX/antimicrox#463] [Ideas]
Hey, I just firstly wanna say how impressive this application is, The calibration section in particular very impressive.. way more than microsoft's (lol tiny flaccid) lazy ass would ever do! :)

But, I've been encountering an issue where I can't seem to disable the game from reading it's assigned gamepad buttons, and i cant disable them. I mean I would prefer to use this app to take care of all of that. And in this case the game is reading the game pad at the same time(before) or so of your app and so its input is handled ever so faster.

Is it possible to block handled gamepad buttons in this? I mean I did something like this in C# but C++ is a whole new monster.
Probably terrible, and you probably don't need this, but:

```csharp
        protected override void WndProc(ref Message msg)
        {
            switch (msg.Msg)
            {
                case Hacking.WM_HOTKEY:  // Is it our hotkey?
                    if ((short)msg.WParam == HotkeyID)
                    {
                        HotkeyAction();
                        //Uncomment the next line to send the hotkey to the game as well.
                        //base.WndProc(ref msg);
                    }
                    break;
                default:
                    // No it wasnt, send the keystroke to the game.
                    base.WndProc(ref msg);
                    break;
            }
        }
```

What I'm asking is if we can opt to turn off  base.WndProc(ref msg);

**Comment by pktiuk** [comment]
For now there is no such an option, but there are some workarounds described in our wiki https://github.com/AntiMicroX/antimicrox/wiki/Methods-for-Disabling-Joystick-Reading-in-Linux

I am not sure whether is easily doable on Windows (and how to do it without hacks on Linux).
